connectors/es/sink.py

```diff
@@ -913,3 +938,20 @@ async def async_bulk(
             enable_bulk_operations_logging=enable_bulk_operations_logging,
         )
         self._sink_task = asyncio.create_task(self._sink.run())
+        self._sink_task.add_done_callback(functools.partial(self.sink_task_callback))
+
+    def sink_task_callback(self, task):
+        if task.exception():
+            self._logger.error(
+                f"Encountered an error in the sync's Sink: {task.get_name()}: {task.exception()}",
+                exc_info=True,
+            )
+            self._sink_error = task.exception()
+
+    def extractor_task_callback(self, task):
+        if task.exception():
+            self._logger.error(
+                f"Encountered an error in the sync's Extractor: {task.get_name()}: {task.exception()}",
+                exc_info=True,
+            )
+            self._extractor_error = task.exception()
```

Comment:
These feel kinda redundant, but since these tasks are called without ConcurrentTasks, there's nothing to log if there was an exception in these that wasn't part of their internal ConcurrentTasks workloads.

Reply:
I no longer think these are redundant, and they're actually critical for making sure that some niche edge case in our framework code doesn't lead to silent failures.